\documentclass[a4paper]{article}

\usepackage{fixltx2e}
\usepackage{hyperref}
\usepackage[T1]{fontenc}  
\usepackage[none]{hyphenat}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{tabularx,ragged2e,booktabs,caption}
\usepackage{amsmath}
\usepackage[left=3cm,right=2.5cm,top=2.5cm,bottom=2.5cm]{geometry}
\usepackage{float}
\usepackage[linktoc=all]{hyperref}
\usepackage{titlesec}
\usepackage{graphicx}
\definecolor{beaublue}{rgb}{0.74, 0.83, 0.9}
\graphicspath{{/Users/alba/Desktop/Rpackages/comorbidity/Data/} }
\hypersetup{
    colorlinks,
    citecolor=blue,
    filecolor=blue,
    linkcolor=blue,
    urlcolor=blue
}

\begin{document}
%\SweaveOpts{concordance=TRUE}


\newcommand{\subsubsubsection}[1]{\vspace{0.10cm}\noindent\textbf{#1}\vspace{0.10cm}}

\title{\texttt{comoRbidity}: an R package for the systematic analysis of disease comorbidities}

\author{Alba Gutierrez-Sacristan \and Alex Bravo \and Alexia Giannoula \and Miguel A. Mayer \and  Ferran Sanz \and Laura I. Furlong}

\maketitle
\tableofcontents
\newpage

\section{Introduction}

\noindent The term comorbidity refers to the coexistence or presence of multiple diseases or disorders in relation to a primary disease or disorder in a patient \cite{REF1, REF2}. Clinical and epidemiological studies indicate that disease comorbidities have a great impact on the evolution of health status, selection of appropriate treatments and health system costs \cite{REF3, REF4}. Understanding comorbidities and their etiology is key to identify new preventive and therapeutic strategies.\\ 

\noindent The goal of the \texttt{comoRbidity} R package is to provide a general overview of the disease comorbidities according to an index diseases, from both, clinical and molecular perspective. Analysing patient-level clinical data allows to identify significant comorbidities, while exploring the gene-disease associations will allow to understand the mechanisms underlying comorbidities. \\

\noindent The \texttt{comoRbidity} R package is a user-friendly disease comorbidity analysis software. It provides different comorbidity measures as well as visualization of comorbidity results (Figure 1). To identify statistically significant comorbidities, the \texttt{comoRbidity} R package uses clinical data, provided by the user, and gene-disease association data based on DisGeNET \cite{REF12} database (www.disgenet.org)(Figure 1A, 1B). Special effort has been made in the results' visualization. Functions for analysis and visualization of the data previous to the comorbidity analysis are available (Figure 1-3). A variety of plots and heatmaps are provided to display the results of the comorbidity analysis (Figure 1-5).\\  

\noindent The tasks that can be performed with the \texttt{comoRbidity} package are the following:
\begin{enumerate}
\item Age and gender analysis of the population diagnosed with the disease of interest. 
\item Clinical comorbidity analysis, based on diagnosis data, in a specific gender and age interval. 
\item Analysis of the comorbidity temporal directionality.
\item Molecular comorbidity analysis, based on shared genes.
\item Visualization of the results in a clear, easily interpretable manner. 
\end{enumerate}
\\

\noindent The \texttt{comoRbidity} package also expedites the integration of comorbidity results with other R packages, and allows the development of complex bioinformatics workflows. In the following sections the specific functions that can be used to address each one of these tasks are presented.\\

\subsection{Installation}

\noindent The package \texttt{comoRbidity} is provided via Bitbucket. To install the \texttt{comoRbidity} R package the user must type the following commands in an R session:

<<bitb, eval=FALSE>>=
library( "devtools" )
install_bitbucket( "ibi_group/disgenet2r" )
install_bitbucket( "ibi_group/comoRbidity" )
library( "comoRbidity" )
@


\begin{figure}[p]
    \makebox[\linewidth]{
        \includegraphics[width=0.7\linewidth]{figuraComorbidityPackage}
    }
    \caption{Overview framework of comoRbidity package} 
    \label{fig:requirements}
\end{figure}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\newpage

\subsection{Implementation and Limitations}

\noindent The \texttt{comoRbidity} R package works under:
\begin{itemize}
\item \textbf{Linux}
\item \textbf{Mac}
\item \textbf{Windows} Note that there are some limitations related to two \texttt{comoRbidity} R functions:
\begin{itemize}
\item \texttt{query} function: set \texttt{python} argument to \texttt{FALSE}
\item \texttt{comorbidityAnalysis} function: set \texttt{cores} argument to \texttt{1}
\end{itemize}
\end{itemize}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Clinical Comorbidity}

\noindent The \texttt{comoRbidity} R package identifies disease comorbidities using different statistical tests and metrics. The disease comorbidity analysis can be performed stratifying the population by age and/or gender. More importantly, it allows the user to analyze his/her \textbf{own clinical record data}, and its own definition of comorbidity (e.g. regarding to the time interval considered). In addition, the \texttt{comoRbidity} package allows to perform other analysis like the Sex ratio analysis (section 2.6) and Temporal Directionality of the concomitant diseases (section 2.7).\\

\subsection{Requirements}

\noindent Four files are required for the comorbidity analysis with the \texttt{comoRbidity} package:
\begin{itemize}
\item patientData
\item diagnosisData
\item admissionData
\item indexDiseaseCodes
\end{itemize}
\\

\noindent An example of the data is shown in the next subsections. This data has been obtained from an artificial medical data set of 100000 patients with 361760 admissions (http://EMRbots.org). \\

<<exampleDataPath, echo = FALSE, warning = FALSE, message = FALSE>>=
databasePth <- system.file("extdata", package="comoRbidity")
patientData <- read.delim( paste0(databasePth, "/patientData.txt"), header = TRUE, sep = "\t" )
diagnosisData <- read.delim( paste0(databasePth, "/diagnosisData.txt"), header = TRUE, sep = "\t" )
admissionData <- read.delim( paste0(databasePth, "/admissionData.txt"), header = TRUE, sep = "\t" )
indexDiseaseCode <- read.delim( paste0(databasePth, "/indexDiseaseCode.txt"), header = TRUE, sep = "\t", colClasses = "character" )
@

\subsubsection{Patient Data}
\noindent The \texttt{patientData} file must contain at least three predefined columns named as stated below:
\begin{itemize}
\item \textbf{patient\_id}: a patient identifier, that can be numeric, alphanumeric or a list of characters.
\item \textbf{patient\_gender}: the patient gender is required to perform a comorbidity analysis stratified by gender and for the sex ratio analysis. It can be numeric (\textit{e.g., 0 for one gender and 1 for the other}), character (\textit{e.g., M for male and F for female}) or a list of characters (\textit{e.g., male and female}).
\item \textbf{patient\_dateBirth}: the patient birth date is required to calculate the age of the patient when he/she has been diagnosed with a particular disease, following the format year, month, day (YYYY/MM/DD), separated by any type of character.
\end{itemize}

\noindent If the \texttt{patientData} file does not contain the required columns the following notification message will appear:
<<message1, echo = FALSE>>=
message("Check the patientData file structure. Remember that this
+ file must contain at least three columns with the column
+ names as follows:\n -> patient_id \n -> patient_gender \n -> patient_dateBirth")
@

<<exampleData_patients>>=
head( patientData )
@

\subsubsection{Diagnosis Data}
\noindent The \texttt{diagnosisData} file must contain at least three predefined columns named as stated below:
\begin{itemize}
\item \textbf{patient\_id}: a patient identifier, that can be numeric, alphanumeric or a list of characters. The patient identifier must be the same that the one used in the patientData file. 
\item \textbf{admission\_id}: an identifier related to the admission or visit, which allows to distinguish among different data entries of the same patient in the database. It can be numeric, alphanumeric or a list of characters. 
\item \textbf{diagnosis\_code}: the disease code or codes assigned to an admission or visit (using any vocabulary or standard to identify disease codes). \textbf{Note that the index diseases in the \texttt{indexDiseaseCode} file must use the same standard or vocabulary as the diagnoses in \texttt{diagnosisData} file.}\\
\end{itemize}

\noindent If the \texttt{diagnosisData} file does not contain the required columns the following notification message will appear:
<<message2, echo = FALSE>>=
 message("Check the diagnosisData file structure. Remember that this
                    file must contain at least three columns with the column 
                    names as follows:\n -> patient_id \n -> admission_id \n -> diagnosis_code" )

@

<<exampleData_diagnosis>>=
head( diagnosisData )
@

\subsubsection{Admission Data}
\noindent The \texttt{admissionData} file must contain at least three predefined columns named as stated below:
\begin{itemize}
\item \textbf{patient\_id}: a patient identifier, that can be numeric, alphanumeric or a list of characters. The patient identifier must be the same one that is used in the patientData file. 
\item \textbf{admission\_id}: an identifier related to the admission or visit in which a disease diagnosis was made, that allows to distinguish between different entries of the patient in the database. It can be numeric, alphanumeric or a list of characters. 
\item \textbf{admissionStartDate}: the date of the admission or visit in which the patient was diagnosed with the diseases under study. This information is needed for the directionality analysis. The date has to follow the format year, month, day (YYYY/MM/DD). Any separator charachter is allowed.\\  
\end{itemize}

\noindent If the \texttt{admissionData} file does not contain the required columns the following notification message will appear:
<<message3, echo = FALSE>>=
 message("Check the admissionData file structure. Remember that this
                    file must contain at least three columns with the column 
                    names as follows:\n -> patient_id \n -> admission_id \n -> admissionStartDate")
@

<<exampleData_admission>>=
head( admissionData )
@

\subsubsection{Index Disease Codes}
\noindent The \texttt{indexDiseaseCode} file contains the diseases in which you are interested for the comorbidity analysis. As explained before, the index diseases in the \texttt{indexDiseaseCode} file must be in the same format as the diagnoses in \texttt{diagnosisData} file.\\

\noindent \texttt{indexDiseaseCode} file must contain at least one predefined columns named as stated below:
\begin{itemize}
\item \textbf{Code}: the disease code (in any format). 
\item \textbf{Agg}: this column is not compulsory for performing the comorbidity analysis. It must be included if the user wants to group the index disease codes in a higher category or class (\textif{e.g., the example below shows all index diseases being grouped in two disease categories, depression (Dep) and bipolar disorder (BD)}).  
\end{itemize}  

<<exampleData_codes>>=
head( indexDiseaseCode )
@

\colorbox{beaublue}{\begin{minipage}{14cm}
\noindent \textbf{While the aforementioned columns are required, the files may contain other additional information, as shown in the previous examples. The extra information will no be used by the \texttt{comoRbidity} R package.}\\
\end{minipage}}
\\

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\subsection{comoRbidity objects}

\subsubsection{comorbidity object}
\noindent The \texttt{comorbidity} object is obtained when the \texttt{query} function is applied. This object is used as input for other functions in the package that enable the user to have an overview of the population under study, including age, gender and diagnosis. \texttt{comorbidity} object is also used as input in the function that performs the comorbidity analysis (\texttt{comorbidityAnalysis}).\\

\noindent In summary, \texttt{comorbidity} object is the input for the following functions:
\begin{itemize}
\item \texttt{comorbidityAnalysis} 
\item \texttt{summaryDB}
\item \texttt{populationAge}
\item \texttt{diagnosisUse}
\end{itemize}
\\

\noindent The \texttt{comorbidity} object contains the query information as well as a summary about the results. It shows:
\begin{itemize}
\item the type of search that has been done (\texttt{Search})
\item if the comorbidities will be estimated only between the index diseases (\texttt{Only comorbidities between index diseases: TRUE}), or if they will be estimated among the index disease and all the disorders (Only comorbidities between index diseases: FALSE)
\item if the individual index disease codes are used for the comorbidity study (\texttt{Aggregate the disease codes: FALSE}) or if they are grouped into a higher category (\texttt{Aggregate the disease codes: TRUE})
\item the number of index disorders used as input (\texttt{N. Input Index Diseases})
\item the total number of disorders present in this subset (\texttt{N. Index Diseases Present})
\item the number of concomitant disorders present in the results (\texttt{N. Concomitant Disorders}) 
\item the patients that suffer at least one of the input disorders (\texttt{N. Patients})
\end{itemize}

\noindent \textbf{All the \texttt{comoRbidity} R package objects come with a function called \texttt{extract}. The \texttt{extract} function allows the user to retrieve data stored in the object. The \texttt{extract} function returns a formatted \texttt{data.frame} with the complete set of information obtained from the required data.}\\ 

<<comor_object, echo = FALSE>>=
load(system.file("extdata", "comorbidity.RData", package="comoRbidity"))
@

<<comor_object2, message=FALSE, warning=FALSE>>=
comor_obj
@


\subsubsection{cAnalysis object}

\noindent The \texttt{cAnalysis} object is obtained when \texttt{comorbidityAnalysis} function is applied. This object is used as input for other functions in the package that enable the user to visualize the results in different graphical ways. Moreover, \texttt{cAnalysis} object is used as input for further analysis in the comorbidity results, like the sex ratio analysis and the directionality analysis.\\

\noindent \texttt{cAnalysis} object is the input for the following functions:
\begin{itemize}
\item \texttt{network}
\item \texttt{heatmapPlot} 
\item \texttt{sexRatio}
\item \texttt{directionality}
\end{itemize}
\\

\noindent The \texttt{cAnalysis} object contains the results of the comorbidity analysis and other relevant information for the user. \texttt{cAnalysis} object shows the age interval that has been applied for the analysis (\texttt{Age Min} and \texttt{Age Max}), the gender, the number of patients that belong to this group from the total data (\texttt{Patients in the age and gender interval}) as well as the number of them that suffer the disease of interest (\texttt{Patients diagnosed with the index disease(s)}). Other data such as the disease prevalence and the range values obtained for each parameter estimated to measure the comorbidity are also contained in the \texttt{cAnalysis} object. Finally, the number of comorbidities that pass the \texttt{cutOff} determined by the user are also shown (\texttt{Number of comorbidities}).\\ 

\noindent \textbf{All the \texttt{comoRbidity} R package objects come with a function called \texttt{extract}. The \texttt{extract} function allows the user to retrieve data stored in the object. The \texttt{extract} function returns a formatted \texttt{data.frame} with the complete set of information obtained from the required data.}\\ 

<<cAnalysis_obj, echo=FALSE>>=
load(system.file("extdata", "comorMale.RData", package="comoRbidity"))
@

<<cAnalysis_obj2>>=
comorMale
class(comorMale)
@

<<extractData>>=
comorbidityData <- extract ( comorMale )
head( comorbidityData )
@

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection{Data extraction}

\noindent The first step in order to perform the comorbidity analysis is extracting the data related to the patients diagnosed with the index diseases. These index diseases are determined by the \texttt{indexDiseaseCode} file. \\

\noindent The \texttt{query} function allows the user to extract the data and store it in a \texttt{comorbidity} class object. As input the \texttt{query} function requires:
\begin{itemize}
\item \textbf{\texttt{databasePth}}: determines the path where the three required input files (\texttt{patientData, diagnosisData, admissionData}) are located.
\item \textbf{\texttt{codesPth}}: determines the path where the file with the index diseases is located (\texttt{indexDiseaseCode}). 
\item \textbf{\texttt{admissionDateSep}}: determines what separator symbol is used in the admission date. 
\item \textbf{\texttt{birthDateSep}}: determines what separator symbol is used in the birth date. 
\end{itemize}

\noindent \begin{minipage}{\linewidth}
\small{
  \centering
  \captionof{table}{Optional arguments for data extraction} \label{tab:optional_arg} 
  \begin{tabular}{p{2cm}|p{1.5cm}|p{8cm}}
    \hline
aggregatedDis argument  &  intraCodes argument  &   Description \\
\hline
FALSE  & FALSE  & Data extraction is done using the Codes column from the index disease file. The comorbidities will be estimated between the index diseases and the rest of diseases that the patient has suffered (\textit{e.g., \texttt{ff}}). \\
TRUE   & FALSE  & Data extraction is done using the Agg column from the index disease file,  that groups the diseases in a higher class. The comorbidities will be estimated between the index diseases and the rest of diseases that the patient has suffered (\textit{e.g., \texttt{aggQuery}}).\\
FALSE  & TRUE   &  Data extraction is done using the Codes column from the index disease file. Comorbidities will be estimated only between the index diseases (\textit{e.g., \texttt{queryIntra}}).\\
TRUE   & TRUE   &  Data extraction is done using the Agg column from the index disease file, that groups the diseases in a higher class. Comorbidities will be estimated only between the index diseases (\textit{e.g., \texttt{aggQueryIntra}}).\\
\hline
\end{tabular} \par
}
\end{minipage}
\\

\noindent As a result, a \texttt{comorbidity} object is obtained. This object will contain those patients that have been diagnosed with at least one of the index diseases presented in the \texttt{indexDiseaseCode} file, and the data related to them, according to the options selected in the \texttt{query} function.\\

\colorbox{beaublue}{\begin{minipage}{12cm}
\noindent \textbf{Note that the \texttt{query} function has an optional argument, \texttt{python}, that by default is \texttt{FALSE}, but that can be changed to \texttt{TRUE} to run the \texttt{query} function faster by using python script. In order to use this option it is necesary to have python installed in your computer. Any python release can be used (import sys and import datetime are required).}
\end{minipage}}
\\

\noindent In the following we illustrate the different options for data extraction shown in Table 1.\\ 
\noindent \textbf{A. aggregatedDis = FALSE & intraCodes = FALSE (Default option)}\\
\noindent The \texttt{databasePth} argument should contain the location of the folder that contains the input files. As an example, it will be used the path where example data is located.
<<paths1 >>=
databasePth <- system.file("extdata", package="comoRbidity")
diagnosticCodes <- system.file("extdata", package="comoRbidity")
@

\noindent The user should indicate his own path following the next structure:
<<pathEx, eval=FALSE>>=
databasePthEx <- "/home/user/.../..."
diagnosticCodesEx <- "/home/user/.../..."
@

<<paths >>=
ff <- query( databasePth      = databasePth, 
             codesPth         = diagnosticCodes, 
             admissionDataSep = "-", 
             birthDataSep     = "-"
             )
ff
@

\\
\noindent \textbf{B. aggregatedDis = TRUE & intraCodes = FALSE}
<<aggregateEx, message=FALSE, warning=FALSE>>=
aggQuery <- query( databasePth      = databasePth, 
             codesPth         = diagnosticCodes, 
             admissionDataSep = "-", 
             birthDataSep     = "-", 
             aggregatedDis    = TRUE
             )
aggQuery
@
\\

\noindent \textbf{C. aggregatedDis = FALSE & intraCodes = TRUE}
<<IntraEx, message=FALSE, warning=FALSE>>=
queryIntra <- query( databasePth      = databasePth, 
             codesPth         = diagnosticCodes, 
             admissionDataSep = "-", 
             birthDataSep     = "-",
             intraCodes       = TRUE, 
             aggregatedDis    = FALSE
             )
queryIntra
@
\\

\noindent \textbf{D. aggregatedDis = TRUE & intraCodes = TRUE}

<<aggregateIntraEx, message=FALSE, warning=FALSE>>=
aggQueryIntra <- query( databasePth      = databasePth, 
             codesPth         = diagnosticCodes, 
             admissionDataSep = "-", 
             birthDataSep     = "-",
             intraCodes       = TRUE, 
             aggregatedDis    = TRUE
             )
aggQueryIntra
@
\\

\subsection{Overview of the clinical data}

\subsubsection{Summary DB}

\noindent The \texttt{comoRbidity} R package allows the user to analyze and characterize the population under study, by using the \texttt{summaryDB} function.\\

\noindent As an input, the \texttt{summaryDB} function requires:
\begin{itemize}
\item \texttt{input}: a \texttt{comorbidity} object, obtained after applying the \texttt{query} function. 
\item \texttt{maleCode}: the symbol which denotes males in users' database (\textif{e.g., 0, M, Male...etc})
\item \texttt{femaleCode}: the symbol which denotes females in users' database (\textit{e.g., 1, F, Female...etc})
\end{itemize}

\noindent The output of the \texttt{summaryDB} function is a plot with three different graphics (Figure \ref{fig:summaryDB}):
\begin{itemize}
\item A barplot with the age distribution of the patients diagnosed with the disease of interest. 
\item A boxplot showing the age distribution by gender. 
\item A pie chart representing the gender distribution.  
\end{itemize}

\begin{figure}[H]
\begin{center}
<<summaryDB, fig=TRUE, height=14, width=14 >>=
summaryDB(input = ff, 
          maleCode = "Male", 
          femaleCode ="Female")
@
\end{center}
\caption{Summary plot containing: age distribution, age distribution by gender and gender distribution.} 
\label{fig:summaryDB}
\end{figure}

\noindent Note that the color of the barplot, the boxplot and the pie chart can be changed adding the next arguments to the \texttt{summaryDB} function:\\
\begin{itemize}
\item \textbf{\texttt{barColor}}: By default \texttt{"darkblue"}. It defines the bar color in the age distribution bar-plot.
\item \textbf{\texttt{femaleColor}}: By default \texttt{"gold"}. It defines the heatmap color for the female value.
\item \textbf{\texttt{maleColor}}: By default \texttt{"lightblue"}. It defines the color for male values.
\end{itemize}

\subsubsection{Population analysis based on the disease under study}

\noindent Expanding on the age analysis, the \texttt{comoRbidity} R package also allows to analyze the age distribution of patients diagnosed with the index disease(s) compared with all the patients of the database. The age of diagnosis with the index disease is the age of the patient when he/she is diagnosed by the first time with that disease.  For the rest of patients, the age considered is the one at the first admission of the patient in the database.\\

\noindent As an input, \texttt{populationAge} function requires:
\begin{itemize}
\item \textbf{\texttt{input}}: a \texttt{comorbidity} object, obtained with the \texttt{query} function. 
\item \textbf{\texttt{codesPth}}: determines the path where the file with the index diseases is located (\texttt{indexDiseaseCode}). 
\item \textbf{\texttt{databasePth}}: determines the path where the three required input files (\texttt{patientData, diagnosisData, admissionData}) are located.
\end{itemize}

\noindent The \texttt{populationAge} function has four additional arguments, that are optional for the user:
\begin{itemize}
\item \textbf{\texttt{type}} argument allows the user to select the output barplot. By default the type is \texttt{"together"} (Figure \ref{fig:popAgeTogether}), but it can be set  to \texttt{"separate"} (Figure \ref{fig:popAgeSeparate}). 
\item \textbf{\texttt{interactive}} argument allows to create an interactive barplot, that show you the specific information of each bar in the barplot in an interactive way.
\item \textbf{\texttt{disorderColor}} argument allows to define the bar color representing the disease patients. By default gold.
\item \textbf{\texttt{allColor}} argument allows to define the bar color representing the whole population. By default blue. 
\end{itemize}


\noindent The results of the population analysis can be visualized together (Figure \ref{fig:popAgeTogether}) or separately. \textbf{Note that the patient age in the whole population is estimated taking into account the first admission date of the patient to the database while the age for index disorder patients is estimated taking into account the first admission date in which the patient has been diagnosed with the index disease.}

\begin{figure}[H]
\begin{center}
<<populationAgeTogether, echo=TRUE, fig=TRUE, out.width='4in' >>=
populationAge ( input       = ff, 
                codesPth    = diagnosticCodes, 
                databasePth = databasePth, 
                type        = "together", 
                interactive = FALSE)
@
\end{center}
\caption{Barplot representing the age-distribution of the patients suffering the disorder of interest vs all population age.} 
\label{fig:popAgeTogether}
\end{figure}

\begin{figure}[H]
\begin{center}
<<populationAgeSeparate, echo=TRUE, fig=TRUE, out.width='4in'>>=
populationAge ( input       = ff, 
                codesPth    = diagnosticCodes, 
                databasePth = databasePth, 
                type        = "separate", 
                interactive = FALSE)
@
\end{center}
\caption{Separate barplot comparing the age-distribution of the patients diagnosed with the disorder of interest vs all population age.} 
\label{fig:popAgeSeparate}
\end{figure}

\subsubsection{Disease Prevalence}

\noindent The \texttt{comoRbidity} R package allows the user to calculate the disease prevalence. In order to obtain this information, the user can apply the \texttt{diseasePrevalence} function.\\

\noindent As a input, the \texttt{diseasePrevalence} function requires:
\begin{itemize}
\item \texttt{input}: a \texttt{comorbidity} object, obtained after applying the \texttt{query} function. 
\item \texttt{maleCode}: the symbol which denotes males in users' database (\textif{e.g., 0, M, Male...etc})
\item \texttt{femaleCode}: the symbol which denotes females in users' database (\textit{e.g., 1, F, Female...etc})
\item \textbf{\texttt{databasePth}}: determines the path where the three required input files (\texttt{patientData, diagnosisData, admissionData}) are located.
\end{itemize}

\noindent The output of the \texttt{diseasePrevalence} function is a barplot showing the disease prevalence in the entire population and the disease prevalence according to gender (Figure \ref{fig:diseasePrevalence}):

\begin{figure}[H]
\begin{center}
<<diseasePrevalence, fig=TRUE, height=14, width=14 >>=
diseasePrevalence(input = ff, 
          maleCode = "Male", 
          femaleCode ="Female", 
          databasePth = databasePth)
@
\end{center}
\caption{Barplot showing the disease prevalence in all the population and also, according to the gender. } 
\label{fig:diseasePrevalence}
\end{figure}

\noindent Note that the color of the barplot can be changed adding the \textbf{\texttt{barColor}} argument to the \texttt{diseasePrevalence} function. By default \texttt{"blue"}. 

\subsubsection{Code Use}

\noindent When studying a disorder that is defined by more than one diagnosis code, the \texttt{comoRbidity} R package allows to analyze the percentage of use of each one of the index disease codes (Figure \ref{fig:codeUse}). 

\noindent As an input, \texttt{diagnosticUse} function requires:
\begin{itemize}
\item \texttt{input}: a \texttt{comorbidity} object, obtained with the \texttt{query} function. \item \textbf{\texttt{codesPth}}: determines the path where the file with the index diseases is located (\texttt{indexDiseaseCode}). 
\end{itemize}

\noindent The \texttt{diagnosticUse} function has two more arguments that are optional for the user:
\begin{itemize}
\item \texttt{cutOff} argument allows the user to select those index disease codes that will be represented in the output barplot. By default the \texttt{cutOff} is set as 0, but it can be set to any other percentage value. 
\item \texttt{interactive} argument allowsthe user to create an interactive barplot, which shows the specific information of each bar in the barplot in an interactive way. By default the \texttt{interactive} argument is set as \texttt{FALSE}. 
\end{itemize}

\begin{figure}[H]
\begin{center}
<<codeUse, fig=TRUE, scale=0.5, out.width='4in' >>=
diagnosticUse( codesPth     = diagnosticCodes, 
               input        = ff, 
               cutOff       = 0,
               interactive  = FALSE
               )
@
\end{center}
\caption{Index disease codes usage in percentage} 
\label{fig:codeUse}
\end{figure}

\noindent Note that the color of the barplot can be changed adding the \textbf{\texttt{barColor}} argument to the \texttt{diagnosticUs}e function. By default \texttt{"darkblue"}.\\ 


\noindent If \texttt{aggregatedDis} argument has been set to \texttt{TRUE}, the graphic will show the percentage of use of each disease category, in this case bipolar disorder (BD) and depression (Dep) (Figure \ref{fig:codeUseAgg}).\\

\begin{figure}[H]
\begin{center}
<<codeUseAgg, fig=TRUE, out.width='4in'>>=
diagnosticUse( codesPth     = diagnosticCodes, 
               input          = aggQueryIntra, 
               cutOff       = 0,
               interactive  = FALSE
               )
@
\end{center}
\caption{Index disease codes usage in percentage when aggregation has been performed} 
\label{fig:codeUseAgg}
\end{figure}



\subsection{Clinical Comorbidity Analysis}

\noindent Once we have a general overview of the patients diagnosed with the index diseases in the database and the main characteristics of this population, the next step is to perform the comorbidity analysis.\\

\noindent The user can estimate the statistically significant comorbidities by applying the \texttt{comorbidityAnalysis} function to the \texttt{comorbidity} object previously generated with the \texttt{query} function.\\

\subsubsection{Comorbidity Measurements}

\noindent The \texttt{comoRbidity} R package estimates several measures to determine if two diseases are comorbid in the population under study: Risk Ratio (RR), Odds Ratio (OR), Comorbidity score, Phi value. In addition, a Fisher exact test can be applied to determine the significance of the associations. \\

\noindent \textbf{Relative risk or Risk Ratio (RR)} The Relative Risk or Risk Ratio (RR) expresses the relationship between disease A and disease B as a ratio measure of effect (or risk) on the disease prevalence.  If we consider that disease A is the outcome variable and disease B the exposure variable, we can ask if a diagnosis of disease B increases the risk of having a diagnosis of disease A. In other words, if disease A co-occurs with disease B more frequently than expected as if they were independent in this population. 

\noindent The RR is estimated as the fraction between the number of patients diagnosed with both diseases and random expectation based on disease prevalence, as described in \cite{REF2}: 

\begin{equation}
\label{eq:solve}
RR_{AB} = \cfrac{C_{AB} N}{P_A P_B}
\end{equation}

\noindent where \textit{C\textsubscript{AB}} is the number of patients diagnosed with both diseases, \textit{N} is the total number of patients in the population and \textit{P\textsubscript{A}} and \textit{P\textsubscript{B}} are the prevalences of diseases A and B.

\noindent The RR can take the following values:
\begin{itemize}
\item RR = 1: Disease A and disease B are independent regarding prevalence.
\item RR > 1: Disease A and disease B co-occur more often than expected by chance.
\item RR < 1: Disease A and disease B co-occur less often than expected by chance.
\end{itemize}
\\

\noindent \textbf{Odds ratio} The odds ratio represents the increased chance that someone suffering disease A will have the comorbid disorder B. It shows the extent to which suffering a disorder increases the risk of developing another illness or disorder. The odds ratio is derived from a comparison of rates of the illness among individuals who do and do not exhibit the factor of interest. A statistically significant odds ratio (significantly different from 1.00 at the .05 level) indicates an appreciable risk associated with a particular factor. For example, an odds ratio of 2.00 indicates a doubled risk of the appearance of the disorder. The 95\% confidence interval is also shown in the output resultant table.\\

\begin{equation}
\label{eq:solve}
OR_{AB} = \cfrac{C_{AB} H}{C_A C_B}
\end{equation}

\noindent where \textit{C\textsubscript{AB}} is the number of patients diagnosed with both diseases, \textit{H} is the number of patients without any of the diseases and \textit{C\textsubscript{A}} and \textit{C\textsubscript{B}} are the number of patients diagnosed with diseases A and B respectively.\\



\noindent \textbf{Comorbidity score} This score is defined in Roque et al. as follows \cite{REF14}:

\begin{equation}
\label{eq:solve}
comorbidity score =log_2 \left( \cfrac{observed + 1}{expected + 1} \right)
, expected = \cfrac{P_{A} P_{B}}{N}
\end{equation}

\noindent where \textit{observed} stands for the number of patients diagnosed with both diseases (disease A and disease B), and \textit{expected} is estimated based on the prevalence of each disease (prevalence of disease A, \textit{P\textsubscript{A}}, multiplied by the prevalence of disease B, \textit{P\textsubscript{B}}, and divided by the total number of patients, \textit{N}). A pseudocount of 1 is added to correct bias of the Comorbidity score towards low prevalent diseases. Since the logarithm is applied, a comorbidity score of 1.0 means that the observed comorbidities are higher than two fold (approximately) than expected.\\

\noindent \textbf{Phi value (Pearsons correlation for binary variables)} measures the robustness of the comorbidity association. It can be expressed mathematically as:

\begin{equation}
\label{eq:solve}
\phi_{AB} = \cfrac{C_{AB} N - P_{A} P_{B}}{\sqrt{P_{A} P_{B}(N - P_{A})(N - P_{A})}}
\end{equation}

\noindent where \textit{N} is the total number of patients in the population, \textit{P\textsubscript{A}} i and \textit{P\textsubscript{B}} are prevalences of diseases A and B respectively. \textit{C\textsubscript{AB}} is the number of patients that have been diagnosed with both diseases A and B, and \textit{P\textsubscript{A} P\textsubscript{B}} is the random expectation based on disease prevalence. 
\noindent The Pearson correlation coefficient, can take a range of values from +1 to -1: 
\begin{itemize}
\item $\phi$\textsubscript{AB} = 0: indicates that there is no correlation between the two diseases.
\item $\phi$\textsubscript{AB} > 0: indicates a positive correlation between the two diseases.
\item $\phi$\textsubscript{AB} < 0: indicates a negative correlation.
\end{itemize}
\\


\noindent \textbf{Fisher test} A Fisher exact test for each pair of diseases is performed to assess the null hypothesis of independence between the two diseases. Four groups of patients are defined in order to perform the statistical testing: patients diagnosed with disease A and disease B, patients diagnosed with disease A but not disease B, patients diagnosed with disease B but not disease A and patients not diagnosed with disease A nor disease B. The Fisher exact test is then applied to estimate the p-value for each pair of diseases. The Benjamini-Hochberg \cite{REF13} false discovery rate method ("fdr") is applied to correct for multiple testing by default. However user can select the best correction method for the analysis changing the \texttt{correctionMethod} argument. The adjustment methods include the Bonferroni correction ("bonferroni"), Holm correction ("holm"), Hochberg correction ("hochberg"), Hommel ("hommel") and Benjamini & Yekutieli ("BY").\\

\noindent These measures allow the user to quantify the co-occurrence of disease pairs compared with the random expectation. The user can select the measure and the cut-off value in order to asses disease comorbidity.\\

\colorbox{beaublue}{\begin{minipage}{12cm}
\noindent \textbf{Note that the output data frame  containing all the comoribidity measurements is sorted by a combined rank of all the different values as described in \cite{REF16}. In order to estimate the  combined rank, and similarly to what is used in many ranking procedures, first we rank each different comorbidity measurement, and subsequently we sum the different rank positions for each measurement.}
\end{minipage}}
\\

\noindent \texttt{comorbidityAnalysis} function allows the user to perform the comorbidity analysis and store it in a \texttt{cAnalysis} class object. As input the function requires:
\begin{itemize}
\item \texttt{input}: a \texttt{comorbidity} object obtained after applying the \texttt{query} function.
\item \textbf{\texttt{codesPth}}: determines the path where the file with the index diseases is located (\texttt{indexDiseaseCode}). 
\item \textbf{\texttt{databasePth}}: determines the path where the three required input files (\texttt{patientData, diagnosisData, admissionData}) are located.
\item \texttt{ageRange}: determines what is the age range of interest for performing the comorbidity analysis. By default it is set from 0 to 100 years old. 
\item \texttt{gender}: determine what is the gender of interest for performing the comorbidity analysis. 
\end{itemize}

\noindent Moreover, the \texttt{comorbidityAnalysis} function allows to restrict the results according to the comorbidity measurements values:
\begin{itemize}
\item score
\item correctedPval
\item oddsRatio
\item rr
\item phi
\end{itemize}

\noindent The user can filter the results by applying all the comorbidity measurements that are considered necesary. The cut-off value for these measurements must be numeric. The example below shows a query in which two of the five comorbidity measurements are applied, the \texttt{score} and the \texttt{correctedPval}.\\

<<comorbidityAnalysisFemales, eval=FALSE >>=
comorFemale <- comorbidityAnalysis( input      = ff, 
                              codesPth         = diagnosticCodes, 
                              databasePth      = databasePth, 
                              score            = 0, 
                              correctionMethod = "fdr",
                              correctedPval    = 1, 
                              ageRange         = c( 0, 100 ), 
                              gender           = "Female", 
                              verbose          = FALSE 
                              )
save(comorFemale, file=paste0(databasePth, "comorFemale.RData"))
@

\noindent As a result, a \texttt{cAnalysis} object is obtained. This object contains a summary of the comorbidities that have been found.\\

<<comorFemaleResults>>=
load(system.file("extdata", "comorFemale.RData", package="comoRbidity"))
comorFemale
@

\noindent \textbf{All the \texttt{comoRbidity} R package objects come with a function called \texttt{extract}. The \texttt{extract} function allows the user to retrieve data stored in the object. The \texttt{extract} function returns a formatted \texttt{data.frame} with the complete set of information obtained from the comorbidity analysis results.}\\ 

<<extractData2>>=
comorbidityDataFem <- extract ( comorFemale )
head( comorbidityDataFem )
@


\subsubsection{Clinical Comorbidity Visualization}

\noindent In order to visualize the comorbidity analysis results, \texttt{comoRbidity} package provides two different options:
\begin{itemize}
\item Network: obtained by applying \texttt{network} function. 
\item Heatmap: obtained by applying \texttt{heatmapPlot} function. 
\end{itemize}


\subsubsubsection{Comorbidity Network}
\\
\noindent \texttt {network} function allows the user to visualize the data contained in the \texttt{cAnalysis} object obtained after applying the \texttt{comorbidityAnalysis} function.\\

\noindent As input the \texttt{network} function requires:
\begin{itemize}
\item \textbf{\texttt{input}}: a \texttt{cAnalysis} object obtained after applying the \texttt{comorbidityAnalysis} function. 
\item \textbf{\texttt{databasePth}}: determines the path where the three required input files (\texttt{patientData, diagnosisData, admissionData}) are located.
\item \textbf{\texttt{layout}}: by deafult \texttt{"layout.fruchterman.reingold"}. It can be set to other of the possible igraph layouts. 
\item \textbf{\texttt{selectValue}}: By default \texttt{"score"} variable will be selected. It can be set to any of the other possible variables ('correctedPval', 'odds ratio', 'phi', 'rr').
\item \textbf{\texttt{cutOff}}: By default \code{'0.05'}. The value of the argument can be changed to any other numeric variable, according to the range of the selected value.
\item \textbf{\texttt{npairs}}: By default \code{'0'}. The value of the argument can be changed to  any other numeric variable to show in the network only those comorbidities suffered by at least \texttt{npairs} of patients.
\item \textbf{\texttt{prop}}: Determines the node size proportionality. By default it is set to \texttt{1}. The value of the argument can be changed to any other numeric variable.
\item \textbf{\texttt{title}}: Determines the title of the network figure. By default \texttt{'Comorbidity network'}. 
\item \textbf{\texttt{interactive}} Determines if the output network is interactive or not. By default the \texttt{interactive} argument is set to \texttt{FALSE}. The value of the argument can be changed to \texttt{TRUE}, as a result an interactive network will be obtained.
\end{itemize}


<<networkMale, eval = FALSE >>=
load(system.file("extdata", "comorFemale.RData", package="comoRbidity"))
network ( input = comorFemale, 
          databasePth = databasePth,
          layout = "layout.fruchterman.reingold", 
          selectValue = "score", 
          cutOff = 1.45,
          npairs = 2, 
          prop  = 1,
          title = "Female comorbidity network", 
          interactive = FALSE
        )
@

\noindent As a result, a network is obtained (Figure \ref{fig:femaleNet}). The nodes in pink belong to the disorder of interest, while the blue nodes correspond to the comorbidity disorders. Note that the color of the nodes can be changed by adding the following arguments to the \texttt{network} function:\\
\begin{itemize}
\item \texttt{diseaseColor}: determines the node color for the disorder of interest. By default it is set to \texttt{"pink"}. 
\item \texttt{comorColor}: determines the node color for the comorbid disorders. By default it is set to \texttt{"lightblue"}.
\end{itemize}

\begin{figure}[H]
\begin{center}
<<networkFemale, fig=TRUE, out.width='4in', echo = FALSE >>=
load(system.file("extdata", "comorFemale.RData", package="comoRbidity"))
network ( input = comorFemale, 
          databasePth = databasePth,
          layout = "layout.fruchterman.reingold", 
          selectValue = "score", 
          cutOff = 1.45, 
          npairs = 2,
          prop  = 1,
          title = "Female comorbidity network"
        )
@
\end{center}
\caption{Comorbidity network in female population} 
\label{fig:femaleNet}
\end{figure}

\noindent The \texttt{comoRbidity} package also allows to visualize \texttt{cAnalysis} object in a heatmap (Figure \ref{fig:femaleHeatmap}). The required input is the same as for the \texttt{network} function.\\ 

<<heatmapPlotMale, out.width='4in', eval=FALSE >>=
heatmapPlot( input       = comorFemale,
             selectValue = "score", 
             npairs      = 2,
             cutOff      = 1.45, 
             verbose     = FALSE,
             interactive = FALSE)
@


\noindent Note that the color of the heatmap can be changed adding the next arguments to the \texttt{heatmaPlot} function:\\
\begin{itemize}
\item \textbf{\texttt{lowColor}}: By default \texttt{"#0000FF"}. It defines the heatmap color for the lowest value.
\item \textbf{\texttt{highColor}}: By default \texttt{"yellow"}. It defines the heatmap color for the highest value. 
\end{itemize}

\begin{figure}[H]
\begin{center}
<<heatmapPlotfemale, fig=TRUE, out.width='4in', echo=FALSE >>=
heatmapPlot( input       = comorFemale,
             selectValue = "score", 
             npairs      = 2, 
             cutOff      = 1.45, 
             verbose     = FALSE,
             interactive = FALSE)
@
\end{center}
\caption{Comorbidity heatmap in female population} 
\label{fig:femaleHeatmap}
\end{figure}


\subsection{Sex ratio analysis}

\noindent The \texttt{comoRbidity} package also estimates the sex ratio (SR) parameter. The sex ratio (SR) parameter allows to see if a comorbidity suffered in both, men and women, is equally likely in both genders or if it is more likely in one gender than in another. For a comorbidity A and age group t, SR \eqref{eq:solve} is defined by Klimek et al. \cite{REF15} as follows:\\

\begin{equation*}
\label{eq:solve}
SR(A,B)=log \left( \cfrac{1 + \cfrac{{D_{f}(B)}}{D_{f}(A,B)}}
{1 + \cfrac{{D_{m}(B)}}{D_{m}(A,B)}} \right)
\end{equation*}


\\

\noindent where Dm(f)(B) stands for the number of patients (men or women) suffering disease B in age group t, and Dm(f)(A,B) denotes those patients suffering disease B who also have been diagnosed with a disease A. SR values close to 0 mean that the comorbidity is equally likely for men and women. Positive SR values indicate that the comorbidity is more likely for women, while negative SR values indicate that the comorbidity is more likely in men.\\

\noindent To obtain the sex ratio heatmap, two steps should be followed:\\
\begin{enumerate}
\item Apply the \texttt{sexRatio} function using as input the \texttt{cAnalysis} object obtained for both genders. 
\item Apply the \texttt{heatmapSexRatio} to the previous results. \texttt{interactive} argument can be set to \texttt{TRUE} if an interactive heatmap is required. 
\end{enumerate}

\noindent In the next example, a filter to the SR results has been applied to show only extreme SR values.\\

<<sexRatio, message=FALSE, warning=FALSE >>=
srAnalysis <- sexRatio(female    = comorFemale, 
                       male      = comorMale, 
                       fisherTest = 0 
                       )

srAnalysis <- srAnalysis[as.numeric(srAnalysis$SR) <= -0.5 
                         | as.numeric(srAnalysis$SR) >= 0.5,]
@

\begin{figure}[H]
\begin{center}
<<SRplot, fig=TRUE, out.width='6in'>>=
heatmapSexRatio(srAnalysis, 
                interactive = FALSE
                )
@
\end{center}
\caption{Sex ratio (SR) comorbidities heatmap} 
\label{fig:srHeatmap}
\end{figure}

\noindent As a result a heatmap (Figure \ref{fig:srHeatmap}) is obtained. In red (positive values) we show these comorbidities that are more likely for women, while in blue (negative values) we show those that are more likely for men. These colors can be changed adding the following arguments:\\
\begin{itemize}
\item \textbf{\texttt{maleColor}}  Determines the heatmap color for those comorbidities that
#' are more likely in men than women. By default \texttt{"blue"}. 
\item \textbf{\texttt{femaleColor}} Determines the heatmap color for those comorbidities that
#' are more likely in women than men. By default \texttt{"red"}. 
\end{itemize}


\subsection{Directionality analysis}

\noindent Additionally, the \texttt{comoRbidity} package allows to analyze the temporal directionality of the co-occurring diseases (Figure \ref{fig:directionHeatmap}).\\

\noindent  The temporal direction of disease association (d\textsubscript{A} $\rightarrow${d}\textsubscript{B}\ and \  d\textsubscript{B} $\rightarrow${d}\textsubscript{A}) is assessed for the diagnosis pairs with a significant Bonferroni-corrected p-value. Specifically, the number of patients for whom diagnosis dB follows diagnosis dA or vice versa, is calculated and an exact binomial test is, subsequently, used with a probability of success equal to 0.5. A preferred (significant) direction is determined for those diagnosis pairs that result in binomial tests with p-values < 0.05 and according to the pair that appears more often.\\

\noindent The \texttt{directionality} function allows the user to perform the directionality analysis. As input the function requires:
\begin{itemize}
\item \textbf{\texttt{input}}: an object of class \texttt{cAnalysis}. 
\item \textbf{\texttt{databasePth}}: determines the path where the three required input files (\texttt{patientData, diagnosisData, admissionData}) are located.
\item \textbf{\texttt{minPairs}}: determines the minimum number of patients that must suffer the comorbidity to take them into account for the directionality analysis. By default the minPairs value is set to \texttt{1}. 
\item \textbf{\texttt{gender}}: determines what is the gender of interest for performing the directionality analysis. 
\item \textbf{\texttt{ageRange}}: determines what is the age range of interest for performing the directionality analysis. By default it is set from 0 to 100 years old. 
\item \textbf{\texttt{days}}: determines the number of days of difference needed for considering two diseases as comorbid for the directionality analysis.
\item \textbf{\texttt{dataSep}}: determines the separator symbol used in the admission date. 
\end{itemize}

<<directionality>>=
comorbidityDirection <- directionality( input       = comorFemale, 
                                        databasePth = databasePth, 
                                        gender      = "Female",
                                        ageRange    = c(0,100),
                                        days        = 0,
                                        minPairs    = 1,
                                        dataSep     = "-")
summary(as.factor(comorbidityDirection$result))
@

\noindent As a result a \texttt{data.frame} is obtained. This \texttt{data.frame} contains 6 columns, with the comorbidity disorders and the directionality results in numeric and character format.\\

<<directionalityDataFrame, echo=FALSE>>=
head(comorbidityDirection)
@

\noindent These results can be visualized in a heatmap by applying the \texttt{heatmapDirection} function. As input this function requires the \texttt{data.frame} obtained after applying the \texttt{directionality} function. The \texttt{interactive} argument is also available. By default it is set to \texttt{FALSE}.\\

<<directionHeatmapData, echo = FALSE >>=
test <- comorbidityDirection[1:50,]
test$result[1:10] <- "From A to B"
test$test[1:10] <- -1
test$result[25:40] <- "From B to A"
test$test[25:40] <- 1
test$test[41:50] <- 0
@

\noindent In the previous example none of the comorbidities show a preferred directionality. For this reason and to illustrate the \texttt{heatmapDirection} output we will apply the function to a \texttt{test} object containing all the possible results.\\

\begin{figure}[H]
\begin{center}
<<directionHeatmap, fig=TRUE, out.width='6in'>>=
heatmapDirection(test, 
                 fromAtoBColor    = "darkgreen", 
                 fromBtoAColor    = "orange", 
                 noDirectionColor = "grey",
                 interactive = FALSE)
@
\end{center}
\caption{Comorbidities directionality heatmap} 
\label{fig:directionHeatmap}
\end{figure}

\noindent As a result, a heatmap is obtained (Figure \ref{fig:directionHeatmap}). Note that three possible results can be shown in the heatmap in three different colors:
\begin{itemize}
\item From A to B (by default: green).
\item From B to A (by default: yellow).
\item No directionality (by default: grey).
\end{itemize}

\noindent Note that the color of the heatmap can be changed adding the next arguments to the \texttt{heatmaDirection} function:\\
\begin{itemize}
\item \textbf{\texttt{fromAtoBColor}}: By default \texttt{"darkgreen"}. It determines the heatmap color when the preferred direction is from A to B.
\item \textbf{\texttt{fromBtoAColor}}: By default \texttt{"orange"}. It determines the heatmap color when the preferred direction is from B to A.
\item \textbf{\texttt{noDirectionColor}}: By default \texttt{"grey"}. It determines the heatmap color when there is no preferred direction.
#' By default it is set to \code{"grey"}
\end{itemize}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\newpage


\section{Molecular Comorbidity}

\subsection{Requirements}

\noindent The \texttt{cuiDiseaseList} file is required in order to perform the molecular comorbidity analysis with \texttt{comoRbidity} package. Alternatively, the user can provide as input a file containing gene disease associations. An example of the data is shown below.\\

\subsubsection{Index Disease Codes}
\noindent The \texttt{cuiDiseaseList} file must contain at least two predefined columns named as stated below:
\begin{itemize}
\item \textbf{identifier\_id}: the UMLS identifier of the disease/s of interest. 
\item \textbf{name}: this column is not compulsory for performing the comorbidity analysis. It must be included if the user want to collapse the index disease codes in a higher category. The example below shows all index diseases collapsed in 7 different categories. 
\end{itemize}   

\noindent If the cuiDiseaseList file does not contain the required columns the next message will appear:
<<message, echo = FALSE>>=
message("Check the input file structure. Remember that this
+ file must contain at least two columns with the column
+ names as follows:\n -> identifier \n -> name\n")
@


<<exampleData_codesMolecular>>=
filePth <- system.file("extdata", package="comoRbidity")
cuiDiseaseList <- read.delim( paste0(filePth, "/cuiDiseaseList.txt"), header = TRUE, sep = "\t" )
head(cuiDiseaseList)
@

\colorbox{beaublue}{\begin{minipage}{14cm}
\noindent \textbf{While the aforementioned columns are required, the files may contain other additional information. The extra information will no be used by the \texttt{comoRbidity} R package for the molecular comorbidity analysis.}\\
\end{minipage}}
\\

\subsubsection{External file containing gene disease association}
\noindent The alternative input file must contain at least two columns with the column names as follows:
\begin{itemize}
\item gene
\item diseaseName
\end{itemize}


\subsection{molecular comoRbidity objects}

\subsubsection{molecularComorbidity object}
\noindent \texttt{molecularComorbidity} object is obtained when \texttt{querymolecular} function is applied. This object is used as input for other functions in the package that enable the user to have an overview about their index diseases as well as the genes associated with them. \texttt{molecularComorbidity} object is used as input in the function that performs the molecular comorbidity analysis (\texttt{comorbidityAnalysisMolecular}).\\

\noindent In summary \texttt{molecularComorbidity} object is the input for the functions:
\begin{itemize}
\item \texttt{summaryDiseases} 
\item \texttt{comorbidityAnalysisMolecular}
\end{itemize}

\noindent The \texttt{molecularComorbidity} object contains the query information as well as the summary of the results. It shows:\\
\begin{itemize}
\item The type of search that has been done (\texttt{Search}).
\item If the \texttt{identifier} column is used for the comorbidity study (\texttt{Aggregate the disease codes: FALSE}), or if they are collapsed into a higher category and the \texttt{name} column is used for the comorbidity study (\texttt{Aggregate the disease codes: TRUE}).
\item \texttt{N. Input Diseases}: number of diseases that the user gives as input.
\item \texttt{N. Index Diseases Present}: number of diseases that present associated genes in DisGeNET.
\item \texttt{N. Genes}: number of genes associated to the diseases. 
\end{itemize}
\\

<<mc_objectLoad, echo=FALSE>>=
load(system.file("extdata", "mc.RData", package="comoRbidity"))
@

<<objectExtample_mc>>=
class(mc)
mc
@

\noindent \textbf{All the \texttt{comoRbidity} R package objects come with a function called \texttt{extract}. The \texttt{extract} function allows the user to retrieve data stored in the object. The \texttt{extract} function returns a formatted \texttt{data.frame} with the complete set of information obtained from the required data.}\\ 

<<objectExtract_mc>>=
head(extract(mc))
@


\subsubsection{molecularcAnalysis object}
\noindent \texttt{molecularcAnalysis} object is obtained when \texttt{comorbidityAnalysisMolecular} or \texttt{dataframe2molecularcomorbidity} functions  are applied. This object is used as input for other functions in the package that enable the user to visualize the results in different graphical ways.\\

\noindent \texttt{molecularcAnalysis} object is the input for the following functions:
\begin{itemize}
\item \texttt{network} 
\item \texttt{heatmapPlot} 
\end{itemize}
\\

\noindent \texttt{molecularcAnalysis} object contains the results of the comorbidity analysis and other relevant information for the user. \texttt{molecularcAnalysis} object shows the gene overlap interval (minimum and maximum value of the gene overlap for the disease comorbidities, Overlap Min and Overlap Max) as well as the Jaccard Index interval (\texttt{Jaccard Min} and \texttt{Jaccard Max}). Other data such as if the p-value has been estimated or not is also shown (\texttt{P-value}). Finally, the number of comorbidities found based on the genes shared between diseases are also shown (\texttt{Number of comorbidities}).\\

<<mcAnalysis_objectLoad, echo=FALSE>>=
load(system.file("extdata", "mcAnalysis.RData", package="comoRbidity"))
@

<<objectExtample_mcAnalysis>>=
class(mcAnalysis)
mcAnalysis
@

\noindent \textbf{All the \texttt{comoRbidity} R package objects come with a function called \texttt{extract}. The \texttt{extract} function allows the user to retrieve data stored in the object. The \texttt{extract} function returns a formatted \texttt{data.frame} with the complete set of information obtained from the required data.}\\ 

<<objectExtract_mcAnalysis>>=
head(extract(mcAnalysis))
@

\subsection{Data extraction}
\noindent The first step to perform the molecular comorbidity analysis is to extract the gene-disease association data for the index diseases. This information is extracted from DisGeNET database (http://disgenet.org).\\

\noindent \texttt{querymolecular} function allows the user to extract the genes associated to the index diseases, based on DisGeNET data, and store it in a \texttt{molecularComorbidity} class object. As input the function requires:
\begin{itemize}
\item \textbf{\texttt{filePth}}: determines the file name with the complete path where the file with disorders of interest is located.
\item \textbf{\texttt{unify}}: the default value is set to \texttt{FALSE}. If the argument is set to \texttt{TRUE}, the \texttt{name} colum from the cui disease file will be selected for performing the comorbidity analysis. 
\item \textbf{\texttt{database}}: the default value is set to \texttt{'CURATED'}. User can select any of the databases available in DisGeNET (Table \ref{tab:disgenet_databases}).
\item \textbf{\texttt{score}}: by default it is set to \texttt{(">", 0)}. It means that all the data available in DisGeNET will be used for the comorbidity analysis. For detailed information about DisGeNET score:\\
\textit{http://disgenet.org/web/DisGeNET/menu/dbinfo#score}.\\
\end{itemize}
\\

\noindent \begin{minipage}{\linewidth}
\small{
  \centering
  \captionof{table}{Source databases included in DisGeNET} \label{tab:disgenet_databases} 
  \begin{tabular}{|l|l|}
    \hline
Name             & Description                                                \\
    \hline
\texttt{CTD\textsubscript{human}}  & The Comparative Toxicogenomics Database, human data \\
\texttt{UNIPROT}    & The Universal Protein Resource                     \\
\texttt{CLINVAR}    & ClinVar, public archive of relationships among sequence variation and human phenotype    \\
\texttt{ORPHANET}    & The portal for rare diseases and orphan drugs         \\
\texttt{GWASCAT}    & The NHGRI-EBI GWAS Catalog                               \\
\texttt{PSYGENET}    & Psychiatric disorders Gene association NETwork          \\
\texttt{HPO}    & Human Phenotype Ontology                     \\
\texttt{GAD}        & The Genetic Association Database    \\
\texttt{CTD\textsubscript{mouse}}  & The Comparative Toxicogenomics Database, \textit{Mus musculus} data \\
\texttt{CTD\textsubscript{rat}}    & The Comparative Toxicogenomics Database, \textit{Rattus norvergicus} data                               \\
\texttt{MGD}        & The Mouse Genome Database  \\
\texttt{RGD}        & The Rat Genome Database    \\
\texttt{LHGDN}      & Literature-derived human gene-disease network generated by text mining Entrez's GeneRIFs  \\
\texttt{BEFREE}     & Data from text mining medline abstracts using the BeFree System  \\
\texttt{CURATED}    & Human curated sources:(CTD\textsubscript{human}), ClinVar, GWASCAT, UniProt, Orphanet, PsyGeNET, and  HPO\\
\texttt{PREDICTED}  & All data from animal models: CTD\textsubscript{rat}, RGD, CTD\textsubscript{mouse}, MGD   \\
\texttt{ALL}        & All previous data sources                                       \\
    \hline
  \end{tabular} \par
}
\end{minipage}
\\

\noindent As a result, a \texttt{molecularComorbidity} object is obtained. This object contains all the gene-disease associations for the index diseases available in DisGeNET according to the user database and score selection.\\

<<mcQuery, message=FALSE, warning=FALSE>>=
diseaseCodes <- paste0(filePth, "/cuiDiseaseList.txt")
mc <- querymolecular ( filePth = diseaseCodes, 
                       unify = FALSE, 
                       database = "CURATED", 
                       score = c(">", 0))
mc
@

\subsection{Overview of the gene-disease data}

\subsubsection{Genes summary}
\noindent The \texttt{comoRbidity} R package allows the user to analyze and characterize the genes associated to the index diseases. In order to have a general overview about the genes, the \texttt{summaryDiseases} function can be applied, by setting the \texttt{type} argument to \texttt{gene\_barplot}.\\

\noindent As input, the \texttt{summaryDiseases} function requires:
\begin {itemize}
\item \textbf{\texttt{input}}: a \texttt{molecularComorbidity} object, obtained after applying the \texttt{querymolecular} function.
\item \textbf{\texttt{type}}: \texttt{'gene\_barplot'} is selected to perform the gene analysis.
\item \textbf{\texttt{database}}: the DisGeNET database from where the gene information will be retrieved (by default: 'CURATED'). 
\item \textbf{\texttt{interactive}}: Determines if the output barplot is interactive or not. By default the \texttt{interactive} argument is set to \texttt{FALSE}. The value of the argument can be changed to \texttt{TRUE}, as a result an interactive barplot generated with Shiny will be obtained.
\end{itemize}

\noindent As a result, a barplot showing the number of diseases associated to each gene will be displayed. Moreover, some gene attributes like the  Disease Pleiotropy Index (DPI, by default the black dot) and Disease Specificity Index (DSI, by default the red dot) are displayed. Note that the default colors of the DPI and DSI can be changed using the arguments dsiColor and dpiColor. These indexes provide an indication on how specific is a gene with respect to diseases. For example, a DSI close to one means that a gene is annotated to a small number of diseases and therefore is more disease-specific, while a low DSI value (e.g. 0.25) indicates that the gene is associated to a large number of diseases and therefore is less disease-specific. For more information about these indexes:
\\ 
\noindent \textit{http://disgenet.org/web/DisGeNET/menu/dbinfo#specificity}.\\

<<mcQueryVisualization, out.width='6in'>>=
summaryDiseases( input  = mc, 
                  type   = "gene_barplot", 
                  database = "CURATED", 
                 interactive = FALSE
           )
@

\subsubsection{Diseases summary}
\noindent The \texttt{comoRbidity} R package allows the user to analyze and characterize the diseases. In order to have a general overview about the number of genes associated to each one of the index diseases as well as the number of disorders that share some genes with them, the \texttt{summaryDiseases} function can be applied, setting the \texttt{type} argument to \texttt{dis\_barplot}.\\

<<mcQueryVisualization2, out.width='6in'>>=
summaryDiseases( input     = mc, 
                 type      = "dis_barplot", 
                 database  = "CURATED")
@

\noindent Note that the bar colors can be changed adding the next arguments to the \texttt{summaryDiseases} function:\\
\begin{itemize}
\item \textbf{\texttt{assocGeneColor}}: By default \texttt{"#E69F00"}. It defines the bar color that represents the number of associated genes
\item \textbf{\texttt{assocDiseaseColor}}: By default \texttt{"136593"}. It defines the bar color that represents the number of associated diseases.
\end{itemize}

\subsection{Molecular comorbidity analysis}

\noindent Having a general overview about the index diseases and the genes associated with them, the next step is to perform the molecular comorbidity analysis.\\

\noindent The user can estimate the molecular comorbidities by applying the \texttt{comorbidityAnalysisMolecular} function to the \texttt{molecularComorbidity} object previously generated with the \texttt{queryMolecular} function.\\

\subsubsection{Molecular Comorbidity Measurements}
\noindent The comorbidity is estimated from the molecular perspective taking into account the number of genes shared between the diseases according to DisGeNET data. The \texttt{comoRbidity} R package estimates the Jaccard  index and optionally a p-value associated to it.\\

\noindent \textbf{Jaccard Index}
\noindent The Jaccard Index, also known as the Jaccard similarity coefficient, is a statistic measurement used for comparing the similarity of two  sets, and is defined as the size of the intersection divided by the size of the union of the sample sets:

\begin{equation*}
J(A, B) = \frac{\mid A \cap B \mid}{\mid A \cup B \mid}
\end{equation*}

\noindent \texttt{comorbidityAnalysisMolecular} function computes the Jaccard Index as an estimation of the similarity of two diseases based on the number of genes shared between the diseases according to DisGeNET data.\\

\noindent \textbf{P-Value}
\noindent To determine if the association between two diseases as estimated by the Jaccard Index is statistically significant, a bootstrap procedure can be applied. \texttt{comorbidityAnalysisMolecular} function has two optional arguments, \texttt{pValue} and \texttt{nboot}. By applying this function, random gene sets of size n and p (being n, p the number of genes associated to disease 1 and 2, respectively) are sampled from a population of human disease genes obtained from DisGeNET CURATED. These random gene sets (n and p) are then used to compute the Jaccard Index for diseases 1 and 2. This procedure is repeated \texttt{nboot} times. Then the number of times that it has been obtained a Jaccard Index for the random gene sets larger than the observed value of the Jaccard Index is computed and represents its p-value.\\

\noindent The \texttt{comorbidityAnalysisMolecular} function allows the user to perform the molecular comorbidity analysis and store it in a \texttt{molecularcAnalysis} object. As input this function requires:
\begin{itemize}
\item \textbf{\texttt{input}}: a \texttt{molecularComorbidity} object obtained after applying the \texttt{querymolecular} function.
\item \textbf{\texttt{pValue}}: determines if the p-value is estimated or not. By default it is set to \texttt{'FALSE'}. The \texttt{pValue} argument can be set to \texttt{'TRUE'} in order to estimate the P-value associated to each Jaccard Index. 
\item \textbf{\texttt{nboot}}: determines the number of random times that the Jaccard Index is computed using random sets. By default it is set to \texttt{100}. The value of the argument can be changed to any other numeric variable. 
\end{itemize}

\\
<<analysisMC_molecular>>=
mcAnalysis <- comorbidityAnalysisMolecular(input  = mc, 
                                           pValue = TRUE, 
                                           nboot  = 1000
                                           )
@

\noindent  As a result, a \texttt{molecularcAnalysis} object is obtained. The \texttt{molecularcAnalysis} object contains the results of the molecular comorbidity analysis and other relevant information for the user. \texttt{molecularcAnalysis} object shows the gene overlap interval for the disease comorbidities (\texttt{Overlap Min} and \texttt{Overlap Max}) as well as the Jaccard Index interval for the disease comorbidities (\texttt{Jaccard Min} and \texttt{Jaccard Max}). Other data such as if the p-value has been estimated or not is shown (\texttt{P-value}). Finally the number of comorbidities that have gene overlap are also shown (\texttt{Number of comorbidities}).\\ 

<<mcQueryResult_molecular2, echo=FALSE>>=
mcAnalysis
head(comoRbidity::extract(mcAnalysis))
@

\subsubsection{Molecular comorbidity visualization}
\noindent In order to visualize the molecular comorbidity analysis results, the \texttt{comoRbidity} R package provides two different options:
\begin{itemize}
\item Network: obtained by applying the \texttt{network} function.
\item Heatmap: obtained by applying the \texttt{heatmapPlot} function. 
\end{itemize}


\subsubsubsection{Comorbidity Network}
\\
\noindent \texttt {network} function allows the user to visualize the data contained in the \texttt{molecularcAnalysis} object obtained after applying the \texttt{comorbidityAnalysisMolecular} function.\\

\noindent As input the \texttt{network} function requires:
\begin{itemize}
\item \textbf{\texttt{input}}: a \texttt{molecularcAnalysis} object obtained after applying the \texttt{comorbidityAnalysisMolecular} function. 
\item \textbf{\texttt{layout}}: by deafult \texttt{"layout.circle"}. It can be set to other of the possible igraph layouts. 
\item \textbf{\texttt{selectValue}}: By default \texttt{"jaccard"} variable will be selected. It can be set to to the p-value variable ('pval').
\item \textbf{\texttt{cutOff}}: By default \code{'0.05'}. The value of the argument can be changed to any other numeric variable, according to the range of the selected value.
\item \textbf{\texttt{npairs}}: By default \code{'0'}. The value of the argument can be changed to  any other numeric variable to show in the network only those comorbidities in which the gene overlap is equal or greather than the \texttt{npairs} value.
\item \textbf{\texttt{prop}}: Determines the node size proportionality. By default it is set to \texttt{1}. The value of the argument can be changed to any other numeric variable.
\item \textbf{\texttt{title}}: Determines the title of the network figure. By default \texttt{'Comorbidity network'}. 
\item \textbf{\texttt{interactive}} Determines if the output network is interactive or not. By default the \texttt{interactive} argument is set to \texttt{FALSE}. The value of the argument can be changed to \texttt{TRUE}, as a result an interactive network will be obtained.
\end{itemize}

\begin{figure}[H]
\begin{center}
<<mcNetwork, fig = TRUE, out.width='4in'>>=
mcNetwork <- network( input = mcAnalysis ,
                      layout = "layout.circle",
                      selectValue = "jaccard", 
                      title = "Molecular comorbidity network",
                      cutOff = 0.05,
                      prop   = 0.2,
                      diseaseColor = "olivedrab1", 
                      interactive = FALSE, 
)
@
\end{center}
\caption{Molecular comorbidity network} 
\label{fig:molecularNetwork}
\end{figure}


\noindent As a result, a network is obtained (Figure \ref{fig:molecularNetwork}). Note that the color of the nodes can be changed by adding the \texttt{diseaseColor} argument to the \texttt{network} function. The \texttt{diseaseColor} argument is set to \texttt{"pink"} color by default.\\

\subsubsubsection{Comorbidity heatmap}
\noindent The \texttt{comoRbidity} package also allows to visualize \texttt{molecularcAnalysis} object in a heatmap (Figure \ref{fig:mcHeatmap}). The required input is the same as for the \texttt{network} function.\\ 

\begin{figure}[H]
\begin{center}
<<mcHeatmap, fig=TRUE,out.width='4in' >>=
mcHeatmap <- heatmapPlot( input = mcAnalysis , 
             selectValue = "jaccard", 
             cutOff = 0.05,
             interactive = FALSE, 
             lowColor = "#0000FF", 
             highColor = "yellow" )
mcHeatmap
@
\end{center}
\caption{Molecular Comorbidity heatmap} 
\label{fig:mcHeatmap}
\end{figure}

\noindent Note that the color of the heatmap can be changed adding the next arguments to the \texttt{heatmaPlot} function:\\
\begin{itemize}
\item \textbf{\texttt{lowColor}}: By default \texttt{"#0000FF"}. It defines the heatmap color for the lowest value.
\item \textbf{\texttt{highColor}}: By default \texttt{"yellow"}. It defines the heatmap color for the highest value. 
\end{itemize}

\section{Warnings}
\noindent All the functions in the \texttt{comoRbidity} R package first check that the input object belongs to the correct class. If the input is not correct the following message will be shown:\\

<<errorMessage, echo=FALSE>>=
message("Check the input object. Remember that this
                    object must be obtained after applying the query
                    function to your input file. The input object class must
                    be:\"cAnalysis\" or \"molecularcAnalysis\"")
@


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\newpage

\section{Bibliography}

\begin{thebibliography}{9}

\bibitem{REF1}
    Capobianco E, Lio' P
    \textbf{Comorbidity: a multidimensional approach.}
    Trends in molecular medicine
    2013 vol: 19 (9) pp: 515-21 
    
\bibitem{REF2}
    Hidalgo C, Blumm N, Barabasi A, Christakis N
    \textbf{A Dynamic Network Approach for the Study of Human Phenotypes}
    PLoS Computational Biology
    2009 vol: 5 (4) pp: e1000353     

\bibitem{REF3}
    Vogeli  C,  Shields  A,  Lee  T,  Gibson  T,  Marder  W,   Weiss  K,  Blumenthal  D.
  \textbf{Multiple  chronic  conditions: prevalence,  health  consequences,  and  implications  for  quality,  care  management,  and  costs.}
  Journal  of  general internal medicine  2007: 391\-39

\bibitem{REF4}
    Hwang  W,  W.  Weller,  H.  Ireys,  Anderson  G.
  \textbf{Out\-of\-pocket  medical   spending  for  care  of  chronic conditions}
  Health Affairs 2001, Vol. 20, No. 6 

\bibitem{REF5}
    Park  J,  Lee  D,  Christakis  NA,  Barabasi  A.
  \textbf{The  impact  of  cellular  networks  on  disease   comorbidity}
  Mol Syst Biol 2009;5:262
  
\bibitem{REF6}
    Park  S,  Yang  JS,  Shin  YE,  Park  J,  Jang  SK,  Kim  S.
  \textbf{Protein  localization  as a  principal feature  of the  etiology and comorbidity of genetic disease.}
  Mol Syst Biol 2011 May 24;7:494.

\bibitem{REF7}
    Lee  D,  Park  J,  Kay  K,  Christakis  N,  Oltvai  Z,  Barabasi   A.
  \textbf{The  implications  of  human  metabolic  network topology for disease comorbidity.}
  Proceedings of the National Academy of Sciences 2008;105(29):9880\-9885.
  
\bibitem{REF8}
    Le  D,  Kwon  Y. 
  \textbf{The  effect  of  feedback  loops  on  disease  comorbidity  in  human  signaling networks.}
  Bioinformatics 2011 Apr;27(8):1113\-1120.

\bibitem{REF9}
    Barabasi, A. L.
  \textbf{Network medicine\-\-from obesity to the "diseasome}
  N Engl J Med 2007, 357:404\-407

\bibitem{REF10}
    Barabasi,  A.  L.,  N.  Gulbahce,  and  J.   Loscalzo.
  \textbf{Network  medicine:  a  network\-based  approach  to  human diseasome}
  Nat Rev Genet 2011, 12:56\-687

\bibitem{REF11}
    Jakovljevic, Miro and Ostojic, Ljerka.
  \textbf{Comorbidity  and  multimorbidity  in  medicine  today:challenges  and  
opportunities  for  bringing  separated  branches  of  medicine  closer  to  each  other.me}
  Mostariensia, 2013. 18\-287
  
\bibitem{REF12}
    Pinero J, Queralt-Rosinach N, Bravo A, Deu-Pons J, Bauer-Mehren A et. al.
  \textbf{DisGeNET: a discovery platform for the dynamical exploration of human diseases and their genes.}
  Database, 2015 vol: 2015 pp: bav028 

\bibitem{REF13}
    Yoav Benjaminia; Dan Draib; Greg Elmerc; Neri Kafkafid; Ilan Golanib
  \textbf{Controlling the false discovery rate in behavior genetics research}
  Behavioural Brain Research 2001
  doi:10.1016/S0166-4328(01)00297-2
  
\bibitem{REF14}
   Roque, Francisco S and Jensen, Peter B and Schmock, Henriette and Dalgaard, Marlene and Andreatta, Massimo and Hansen, Thomas and S{\o}eby, Karen and Bredkj{\ae}r, S{\o}ren and Juul, Anders and Werge, Thomas and others
  \textbf{Using Electronic Patient Records to Discover Disease Correlations and Stratify Patient Cohorts}
  PLOS Computational Biology 2011
  doi:10.1371/journal.pcbi.1002141

\bibitem{REF15}
    Klimek, Peter and Kautzky-Willer, Alexandra and Chmiel, Anna and Schiller-Fr{\"u}hwirth, Irmgard and Thurner, Stefan
  \textbf{Quantification of diabetes comorbidity risks across life using nation-wide big claims data.}
  PLoS Comput. Biol. 2015
  doi: 0.1371/journal.pcbi.1004125

\bibitem{REF16}
   Gomez-Cabrero, David and Menche, J{\"o}rg and Vargas, Claudia and Cano, Isaac and Maier, Dieter and Barab{\'a}si, Albert-L{\'a}szl{\'o} and Tegn{\'e}r, Jesper and Roca, Josep
  \textbf{From comorbidities of chronic obstructive pulmonary disease to identification of shared molecular mechanisms by data integration}
  BMC bioinformatics 2016

 
\end{thebibliography}

\end{document}